### PR TITLE
Improve OREID payment integration

### PR DESCRIPTION
**Pull request by Alexey1100**
Resolves:
https://www.pivotaltracker.com/story/show/177615677
https://www.pivotaltracker.com/story/show/178529474
https://www.pivotaltracker.com/story/show/178529600
https://www.pivotaltracker.com/story/show/178883783

- Improve OREID stimulus controller
- Integrate OREID stimulus controller into Accounts and Transfer Rules
pages
- Add flash message regarding missing OREID wallet when redirecting to Wallets page
- Fix duplicated records on Accounts page

### CODE REVIEW

File: app/controllers/dashboard/accounts_controller.rb [hunk 1]
```diff
@@ -129,7 +129,7 @@ def accounts_query
       if @project.token&.token_type&.operates_with_account_records?
         @q = @q.joins(:account_token_records)
                .where(account_token_records: { token_id: @project.object.token_id })
-               .includes(account_token_records: [:reg_group])
+               .includes(account_token_records: [:reg_group]).distinct
```
Comment (zhinfuhnya) [inline app/controllers/dashboard/accounts_controller.rb:132]: it's better to move `.distinct` to a new line : )

File: app/controllers/wallets_controller.rb [hunk 2]
```diff
@@ -105,6 +106,10 @@ def wallets_collection
       end
     end
 
+    def set_flash
+      flash.now[:error] = "To sign transactions with ORE ID, please link your ORE ID wallet and make sure it's set as primary." if params[:ore_id_required]
```
Comment (zhinfuhnya) [inline app/controllers/wallets_controller.rb:110]: In my opinion, our controllers have a lot of callbacks, which is not good practice.

I'm in favor of bringing the flash directly into the action:

```
def index
    @wallets = wallets_collection.includes(balances: :token).order(:_blockchain)

    respond_to do |format|
      format.html { flash: { error: 'To sign transactions with ORE ID, please link your ORE ID wallet and make sure it's set as primary.' } if params[:ore_id_required] }
      format.json { render_collection_partial }
    end
  end
```

**Review by zhinfuhnya — DISMISSED** [review]

File: app/controllers/wallets_controller.rb (hunk 2 above)
Comment (Alexey1100) [reply app/controllers/wallets_controller.rb:110]: True, but it's all about balance :)
Params handlers should live in a callbacks I think.

Comment (aquabu) [reply app/controllers/wallets_controller.rb:110]: Let's discuss it and if we want to refactor we can include it in a later commit. The refactoring could be significant if we want to adopt a new pattern.

**Review by aquabu — APPROVED** [review]

**Pull request merged**